docs/aws/audit/elbmonitoring/rules/fms_webacl_resource_policy.mdx

```diff
@@ -23,6 +23,225 @@ CBP
+
+4. In the policy details page, check the "AWS WAF web ACL" field. If the policy owner has specified a WebACLId for the Elastic Load Balancer, it will be displayed here. If this field is empty or not set, then the FMS Policy Owner has not specified a WebACLId.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to execute the commands.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the policies in AWS Firewall Manager (FMS):
+
+   ```
+   aws fms list-policies
+   ```
+   This command will return a list of all FMS policies. Note down the PolicyId of the policy you want to check.
+
+3. Now, use the following command to get the details of the specific policy:
+
+   ```
+   aws fms get-policy --policy-id <PolicyId>
+   ```
+   Replace `<PolicyId>` with the actual PolicyId you noted down in the previous step. This command will return the details of the policy including the WebACLId.
+
+4. Finally, you can use the following command to describe the load balancers and check if the WebACLId is specified in the load balancer:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a list of all the load balancers. Check the WebACLId in the output. If the WebACLId is not specified in the load balancer, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, make sure you have the necessary Python libraries installed. You will need the boto3 library, which is the Amazon Web Services (AWS) SDK for Python. It allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Set up AWS credentials: You need to configure your AWS credentials. You can do this by setting the following environment variables:
+
+   ```python
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   ```
+
+   Alternatively, you can use the AWS CLI to configure your credentials:
+
+   ```bash
+   aws configure
+   ```
+
+3. Write a Python script to check the FMS Policy: Now you can write a Python script that uses the boto3 library to check the FMS Policy. Here is a basic example:
+
+   ```python
+   import boto3
+
+   # Create a client
+   client = boto3.client('fms')
+
+   # Get the policy
+   response = client.get_policy(PolicyId='your_policy_id')
+
+   # Check if the policy specifies a WebACLId
+   if 'Policy' in response and 'SecurityServicePolicyData' in response['Policy'] and 'ManagedServiceData' in response['Policy']['SecurityServicePolicyData']:
+       managed_service_data = json.loads(response['Policy']['SecurityServicePolicyData']['ManagedServiceData'])
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>